### DISCUSSION TITLE
Webview Information

### DISCUSSION THREAD

**Discussion by CGameDev** [JK-Rolling/DuinoI2C_ESP#4] [Ideas]
Is it possible to add more information and configuration in the Webview
1. The current firmware would be nice for users like me with multiple ESPs and need to keep track of whats updated.
2. The current IP Address for users without LCD Display.
3. An option to set the current IP as a static IP, either in Webview or Config portal.
4. 
![espversion](https://github.com/user-attachments/assets/454524c5-b602-4c4a-895d-1673afec9d19)

**Comment by JK-Rolling** [comment]
per discussion in [discord](https://discord.gg/duino-coin-677615191793467402),
summarized below
- static IP enablement from WM
- show `fw_ver`, `ip` on web dashboard
- provide URL to return fw_ver, IP, etc. example <http://myprecious.local/info>
- gh to add a trigger file where the content contain `fw_ver`

I'll need to evaluate the impact on the heap and plan it out

**Reply by CGameDev** [reply]
Sounds like a plan, using power shell to check and update would be nice as well


```
$credentials = New-Object System.Net.NetworkCredential("admin", "admin") #Set Credentials

$getFirmVer = Invoke-WebRequest -Uri http://mwprecious.local/info #Request Current Device Firmeware

$Check = $getFirmVer.Firmware #Get device current firmware

$firmwareUpdate = "DuinoI2C_ESP.ino.d1_mini.400K.ledD7.v0.23.1.bin" #The name of the update to be install OTA

$findchar = $firmwareUpdate.IndexOf("v")

$substring_after = $firmwareUpdate.Substring($findchar+1)

$versionCode = "$substring_after"

$standardizedCode = $versionCode -replace 'bin', ''

$modifiedString = $standardizedCode.Replace('.', '') #removing periods from firmware

Write-Host "New Firmware Update Version Code: " $modifiedString`n

if($Check -gt $modifiedString -or $Check -eq $modifiedString) #checking current firmware against new firmware
{Write-Host "Your Firmware is Up-To-Date"} # Will nor update if device version is equal or newer
else
{Invoke-WebRequest -Uri http://192.168.0.235:54321/firmware -Method Post -Credential $credentials -Body "image=@$firmwareUpdate"} #Update is device version is lower
```
I do a test on the filename.
![image](https://github.com/user-attachments/assets/4c792ae7-d770-448f-9cbb-2df3d35a372a)

**Comment by JK-Rolling** [comment]
Feature implemented from v0.24 and beyond. Thanks for the idea, CGameDev!
